plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/GodotPlugin.kt

```diff
@@ -0,0 +1,111 @@
+    override fun apply(project: Project) {
+        require(project.rootProject == project) { "godot-kotlin plugin can only be applied on the root project!" }
+        project.pluginManager.withPlugin("org.jetbrains.kotlin.multiplatform") {
+            val mpp = project.extensions.getByType<KotlinMultiplatformExtension>()
+            val godot = project.extensions.create<GodotExtension>("godot", project.objects)
+            setupExtensionDefaults(project, godot)
+            setupKotlinPlugin(project, mpp, godot)
+        }
+    }
+
+    private fun setupExtensionDefaults(project: Project, godot: GodotExtension) {
+        with(godot) {
+            platforms(*Platform.values())
+            debug.set(false)
+            cleanupGeneratedFiles.set(true)
+            gdnsDir.set(project.file("src/gdns/kotlin"))
+            entrySourceDir.set(project.buildDir.resolve("godot-entry"))
+        }
+    }
+
+    private fun setupKotlinPlugin(project: Project, mpp: KotlinMultiplatformExtension, godot: GodotExtension) {
+        project.afterEvaluate {
+            // create the main source set
+            val godotMain = mpp.sourceSets.create(MAIN_SOURCE_SET_NAME)
+            val godotTest = mpp.sourceSets.create(TEST_SOURCE_SET_NAME)
+
+            fun KotlinNativeTarget.configureSourceSets(includeEntrySourceDir: Boolean) {
+                compilations.getByName("main").defaultSourceSet {
+                    dependsOn(godotMain)
+                    if (includeEntrySourceDir) {
+                        kotlin.srcDirs += godot.entrySourceDir.get().asFile
+                    }
+                }
+
+                compilations.getByName("test").defaultSourceSet {
+                    dependsOn(godotTest)
+                }
+            }
+
+            // create dummy target and connect it to the main source set
+            // this target is only used for generating entry source files.
+            val os = OperatingSystem.current()
+            val dummyTarget = when {
+                os.isLinux -> mpp.linuxX64("dummyTarget")
+                os.isWindows -> mpp.mingwX64("dummyTarget")
+                os.isMacOsX -> mpp.macosX64("dummyTarget")
+                else -> throw AssertionError("Unsupported operating system: $os")
+            }
+            dummyTarget.configureSourceSets(false)
+            godot.configureTarget(dummyTarget)
+
+            val buildTask = project.tasks.getByName("build")
+
+            // create the targets and connect it to the main source set
+            godot.platforms.get().forEach { platform ->
+                val target = when (platform) {
+                    Platform.LINUX_X64 -> mpp.linuxX64("linuxX64")
+                    Platform.WINDOWS_X64 -> mpp.mingwX64("windowsX64")
+                    Platform.OSX_X64 -> mpp.macosX64("macosX64")
+                    Platform.ANDROID_X32 -> mpp.androidNativeArm32("androidX32")
+                    Platform.ANDROID_X64 -> mpp.androidNativeArm64("androidX64")
```

Comment:
These are the arm targets not the X targets right? So naming is wrong? Also then the "real" X targets are missing?